docs/guide/input-validation.md

```diff
@@ -442,6 +442,381 @@ class EntryForm extends Model
+
+    /**
+     * Validates the specified object.
+     * @param \yii\base\Model $model the data model being validated.
+     * @param array|null $attributes the list of attributes to be validated.
+     * Note that if an attribute is not associated with the validator, or is is prefixed with `!` char - it will be
+     * ignored. If this parameter is null, every attribute listed in [[attributes]] will be validated.
+     */
+    public function validateAttributes($model, $attributes = null)
+    {
+        if (is_array($attributes)) {
+            $newAttributes = [];
+            foreach ($attributes as $attribute) {
+                if (in_array($attribute, $this->attributes) || in_array('!' . $attribute, $this->attributes)) {
+                    $newAttributes[] = $attribute;
+                }
+            }
+            $attributes = $newAttributes;
+        } else {
+            $attributes = [];
+            foreach ($this->attributes as $attribute) {
+                $attributes[] = $attribute[0] === '!' ? substr($attribute, 1) : $attribute;
+            }
+        }
+
+        foreach ($attributes as $attribute) {
+            $skip = $this->skipOnError && $model->hasErrors($attribute)
+                || $this->skipOnEmpty && $this->isEmpty($model->$attribute);
+            if ($skip) {
+                // Skip validation if at least one attribute is empty or already has error
+                // (according skipOnError and skipOnEmpty options must be set to true
+                return;
+            }
+        }
+
+        if ($this->batch) {
+            // Validate all attributes at once
+            if ($this->when === null || call_user_func($this->when, $model, $attribute)) {
+                // Pass array with all attributes instead of one attribute
+                $this->validateAttribute($model, $attributes);
+            }
+        } else {
+            // Validate each attribute separately using the same validation logic
+            foreach ($attributes as $attribute) {
+                if ($this->when === null || call_user_func($this->when, $model, $attribute)) {
+                    $this->validateAttribute($model, $attribute);
+                }
+            }
+        }
+    }
+}
+```
+
+Then we need to create custom validator and use the created trait:
+
+```php
+<?php
+
+namespace app\components;
+
+use yii\validators\Validator;
+
+class CustomValidator extends Validator
+{
+    use BatchValidationTrait;
+}
+```
+
+To support inline validation as well we can extend default inline validator and also use this trait:
+
+```php
+<?php
+
+namespace app\components;
+
+use yii\validators\InlineValidator;
+
+class CustomInlineValidator extends InlineValidator
+{
+    use BatchValidationTrait;
+}
+```
+
+Couple more changes are needed.
+
+First to use our `CustomInlineValidator` instead of default `InlineValidator` we need to override
+[[yii\validators\Validator::createValidator()]] method in `CustomValidator`:
```

Comment:
In other places of the docs (even in the same file) it's used without leading backslash. Used it as an example.